src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -304,109 +298,89 @@
         (str/replace param-name-pattern #(anonymize-leaf % :parameter-name context config))
         (str/replace title-pattern #(anonymize-leaf % :title context config)))))
 
-(defn update-in-nil
-  "Wrapper around update-in that ignores keys with nil"
-  [m [k] f & args]
-  (if (nil? (get m k))
-    m
-    (if args
-      (apply update-in m [k] f args)
-      (apply update-in m [k] f))))
-
-(defn anonymize-resource
-  "Anonymize a resource"
+(defn anonymize-catalog-resource
   [resource context config]
   {:pre  [(resource? resource)]
    :post [(resource? %)]}
-  (let [newcontext {"node"  (get context "node")
+  (let [newcontext {"node" (get context "node")
                     "title" (get resource "title")
-                    "tags"  (get resource "tags")
-                    "file"  (get resource "file")
-                    "line"  (get resource "line")
-                    "type"  (get resource "type")}]
+                    "tags" (get resource "tags")
+                    "file" (get resource "file")
+                    "line" (get resource "line")
+                    "type" (get resource "type")}]
     (-> resource
-        (update-in-nil ["file"]       anonymize-leaf :file newcontext config)
-        (update-in-nil ["line"]       anonymize-leaf :line newcontext config)
-        (update-in     ["parameters"] anonymize-parameters newcontext config)
-        (update-in     ["tags"]       anonymize-tags newcontext config)
-        (update-in     ["title"]      anonymize-leaf :title newcontext config)
-        (update-in     ["type"]       anonymize-leaf :type newcontext config))))
-
-(defn anonymize-resources
-  "Anonymize a collection of resources"
+        (utils/update-when ["file"] anonymize-leaf :file newcontext config)
+        (utils/update-when ["line"] anonymize-leaf :line newcontext config)
+        (update "parameters" anonymize-parameters newcontext config)
+        (update "tags" anonymize-tags newcontext config)
+        (update "title" anonymize-leaf :title newcontext config)
+        (update "type" anonymize-leaf :type newcontext config))))
+
+(defn anonymize-catalog-resources
   [resources context config]
   {:pre  [(coll? resources)]
    :post [(coll? %)]}
-  (map #(anonymize-resource % context config) resources))
+  (map #(anonymize-catalog-resource % context config) resources))
 
-(pls/defn-validated anonymize-resource-event :- resource-event-schema-str
-  "Anonymize a resource event from a report"
-  [event :- resource-event-schema-str
+(defn anonymize-event
```

Comment:
did you intend to remove the validation on these functions? seems like we've lost it in a number of places, but may still be covered at the edges